src/software/backend/wifi_backend.cpp

```diff
@@ -27,7 +27,7 @@ WifiBackend::WifiBackend()
 
 void WifiBackend::onValueReceived(ConstPrimitiveVectorPtr primitives_ptr)
 {
-    primitive_output->sendProto(*createPrimitiveMsg(primitives_ptr));
+    primitive_output->sendProto(*createPrimitiveSetMsg(primitives_ptr));
```

Comment:
```suggestion
    primitive_output->sendProto(*(createPrimitiveSetMsg(primitives_ptr).release()));
```

Reply:
Won't that be a memory leak? We release the unique ptr, so now we have a raw pointer that we're responsible for deleting. We proceed to dereference it and then exit the function, losing the memory associated with it.

Reply:
`sendProto` doesn't take ownership of the pointer, like other `set_allocated_*` proto functions do, so we may mem leak here

Reply:
`sendProto` just takes a const-reference to the object and eventually serializes it into a string that it owns internally, so never takes ownership. I think the original code is ok.